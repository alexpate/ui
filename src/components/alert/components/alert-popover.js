/* @flow */
import React from 'react';
import {Box} from '../../layout';

type TProps = {
  children: React$Element<*>,
  topOffset?: number,
};

export default function AlertPopover(props: TProps) {
  const {children, topOffset = 0} = props;

  return (
    <Box position="fixed" right={32} top={topOffset + 16} maxWidth={400}>
      {React.Children.map(children, child =>
        <Box marginBottom={8}>
          {React.cloneElement(child, {justifyContent: 'left'})}
        </Box>
      )}
    </Box>
  );
}
