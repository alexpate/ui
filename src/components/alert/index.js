/* @flow */
export {default} from './alert';
export {default as AlertPopover} from './components/alert-popover';
