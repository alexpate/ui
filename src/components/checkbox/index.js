/* @flow */
export {default} from './checkbox';
