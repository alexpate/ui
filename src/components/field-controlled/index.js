/* @flow */
export {default} from './field-controlled';
