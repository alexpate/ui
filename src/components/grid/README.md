# Grid

Some low level grid components