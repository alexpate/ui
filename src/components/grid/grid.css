:root {
  --gutter-width: 1rem;
  --outer-margin: 2rem;
  --gutter-compensation: calc((var(--gutter-width) * 0.5) * -1);
  --half-gutter-width: calc((var(--gutter-width) * 0.5));
  --xs-min: 30;
  --sm-min: 48;
  --md-min: 64;
  --lg-min: 75;
  --screen-xs-min: var(--xs-min)em;
  --screen-sm-min: var(--sm-min)em;
  --screen-md-min: var(--md-min)em;
  --screen-lg-min: var(--lg-min)em;
  --container-sm: calc(var(--sm-min) + var(--gutter-width));
  --container-md: calc(var(--md-min) + var(--gutter-width));
  --container-lg: calc(var(--lg-min) + var(--gutter-width));
}

.container {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
}

.container.fluid {
  padding-right: var(--outer-margin, 2rem);
  padding-left: var(--outer-margin, 2rem);
}

.row {
  box-sizing: border-box;
  display: flex;
  flex: 0 1 auto;
  flex-direction: row;
  flex-wrap: wrap;
  margin-right: var(--gutter-compensation, -0.5rem);
  margin-left: var(--gutter-compensation, -0.5rem);
}

.row.reverse {
  flex-direction: row-reverse;
}

.col.reverse {
  flex-direction: column-reverse;
}

.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12 {
  box-sizing: border-box;
  flex: 0 0 auto;
  padding-right: var(--half-gutter-width, 0.5rem);
  padding-left: var(--half-gutter-width, 0.5rem);
}

.col-xs {
  flex-grow: 1;
  flex-basis: 0;
  max-width: 100%;
}

.col-xs-1 {
  flex-basis: 8.33333333%;
  max-width: 8.33333333%;
}

.col-xs-2 {
  flex-basis: 16.66666667%;
  max-width: 16.66666667%;
}

.col-xs-3 {
  flex-basis: 25%;
  max-width: 25%;
}

.col-xs-4 {
  flex-basis: 33.33333333%;
  max-width: 33.33333333%;
}

.col-xs-5 {
  flex-basis: 41.66666667%;
  max-width: 41.66666667%;
}

.col-xs-6 {
  flex-basis: 50%;
  max-width: 50%;
}

.col-xs-7 {
  flex-basis: 58.33333333%;
  max-width: 58.33333333%;
}

.col-xs-8 {
  flex-basis: 66.66666667%;
  max-width: 66.66666667%;
}

.col-xs-9 {
  flex-basis: 75%;
  max-width: 75%;
}

.col-xs-10 {
  flex-basis: 83.33333333%;
  max-width: 83.33333333%;
}

.col-xs-11 {
  flex-basis: 91.66666667%;
  max-width: 91.66666667%;
}

.col-xs-12 {
  flex-basis: 100%;
  max-width: 100%;
}

.col-xs-offset-0 {
  margin-left: 0;
}

.col-xs-offset-1 {
  margin-left: 8.33333333%;
}

.col-xs-offset-2 {
  margin-left: 16.66666667%;
}

.col-xs-offset-3 {
  margin-left: 25%;
}

.col-xs-offset-4 {
  margin-left: 33.33333333%;
}

.col-xs-offset-5 {
  margin-left: 41.66666667%;
}

.col-xs-offset-6 {
  margin-left: 50%;
}

.col-xs-offset-7 {
  margin-left: 58.33333333%;
}

.col-xs-offset-8 {
  margin-left: 66.66666667%;
}

.col-xs-offset-9 {
  margin-left: 75%;
}

.col-xs-offset-10 {
  margin-left: 83.33333333%;
}

.col-xs-offset-11 {
  margin-left: 91.66666667%;
}

.justify-xs-start {
  justify-content: flex-start;
  text-align: start;
}

.justify-xs-center {
  justify-content: center;
  text-align: center;
}

.justify-xs-end {
  justify-content: flex-end;
  text-align: end;
}

.justify-xs-around {
  justify-content: space-around;
}

.justify-xs-between {
  justify-content: space-between;
}

.align-xs-start {
  align-items: flex-start;
}

.align-xs-center {
  align-items: center;
}

.align-xs-end {
  align-items: flex-end;
}

@media screen and (min-width: 768px) {
  .container {
    max-width: var(--container-sm, 46rem);
  }

  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-sm {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-sm-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-sm-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-sm-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-sm-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-sm-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-sm-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-sm-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-sm-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-sm-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-sm-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-sm-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-sm-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-sm-offset-0 {
    margin-left: 0;
  }

  .col-sm-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-sm-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-sm-offset-3 {
    margin-left: 25%;
  }

  .col-sm-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-sm-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-sm-offset-6 {
    margin-left: 50%;
  }

  .col-sm-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-sm-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-sm-offset-9 {
    margin-left: 75%;
  }

  .col-sm-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-sm-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-sm-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-sm-center {
    justify-content: center;
    text-align: center;
  }

  .justify-sm-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-sm-around {
    justify-content: space-around;
  }

  .justify-sm-between {
    justify-content: space-between;
  }

  .align-sm-start {
    align-items: flex-start;
  }

  .align-sm-center {
    align-items: center;
  }

  .align-sm-end {
    align-items: flex-end;
  }
}

@media screen and (min-width: 1024px) {
  .container {
    max-width: var(--container-md, 61rem);
  }

  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-md {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-md-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-md-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-md-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-md-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-md-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-md-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-md-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-md-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-md-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-md-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-md-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-md-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-md-offset-0 {
    margin-left: 0;
  }

  .col-md-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-md-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-md-offset-3 {
    margin-left: 25%;
  }

  .col-md-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-md-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-md-offset-6 {
    margin-left: 50%;
  }

  .col-md-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-md-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-md-offset-9 {
    margin-left: 75%;
  }

  .col-md-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-md-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-md-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-md-center {
    justify-content: center;
    text-align: center;
  }

  .justify-md-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-md-around {
    justify-content: space-around;
  }

  .justify-md-between {
    justify-content: space-between;
  }

  .align-md-start {
    align-items: flex-start;
  }

  .align-md-center {
    align-items: center;
  }

  .align-md-end {
    align-items: flex-end;
  }
}

@media screen and (min-width: 1424px) {
  .container {
    max-width: var(--container-lg, 71rem);
  }

  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-lg {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-lg-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-lg-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-lg-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-lg-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-lg-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-lg-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-lg-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-lg-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-lg-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-lg-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-lg-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-lg-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-lg-offset-0 {
    margin-left: 0;
  }

  .col-lg-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-lg-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-lg-offset-3 {
    margin-left: 25%;
  }

  .col-lg-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-lg-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-lg-offset-6 {
    margin-left: 50%;
  }

  .col-lg-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-lg-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-lg-offset-9 {
    margin-left: 75%;
  }

  .col-lg-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-lg-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-lg-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-lg-center {
    justify-content: center;
    text-align: center;
  }

  .justify-lg-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-lg-around {
    justify-content: space-around;
  }

  .justify-lg-between {
    justify-content: space-between;
  }

  .align-lg-start {
    align-items: flex-start;
  }

  .align-lg-center {
    align-items: center;
  }

  .align-lg-end {
    align-items: flex-end;
  }
}
