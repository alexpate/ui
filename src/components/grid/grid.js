/* @flow */
import React from 'react';
import styled from 'react-emotion';
import {toNumber} from 'lodash';

import {Box, Flex} from '../layout';

/**
 * Some low level grid components
 */
export const Grid = styled(Flex)`
  width: 100%;
  flex-direction: column;
  max-width: ${props => props.theme.grid.gridMaxWidth}px;
  margin-left: auto;
  margin-right: auto;
`;

Grid.displayName = 'Grid';

type TRowProps = {
  children?: any,
  gutters?: 'extra-small' | 'small' | 'medium' | 'large' | 'extra-large',
  collapse?: boolean,
};

export function Row(props: TRowProps) {
  const {children, gutters = 'medium', collapse = false, ...otherProps} = props;

  return (
    <Flex width="100%" flexWrap="wrap" {...otherProps}>
      {React.Children.map(children, (child, i) => {
        const isLast = React.Children.count(children) === i + 1;
        const isFirst = i === 0;

        return child
          ? React.cloneElement(child, {
              paddingLeft: collapse || isFirst ? null : gutters,
              paddingRight: collapse || isLast ? null : gutters,
            })
          : null;
      })}
    </Flex>
  );
}

type TColumnProps = {
  columns?: number,
  children?: any,
  spacing?: 'small',
};

export function Column(props: TColumnProps) {
  const {columns = 12, children, ...otherProps} = props;

  const _width = `${100 / 12 * toNumber(columns)}%`;

  return (
    <Box flexDirection="column" width={_width} {...otherProps}>
      {children}
    </Box>
  );
}
