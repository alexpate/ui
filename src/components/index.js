/**
 * Typography
 */
export { H1 } from './typography/h1';
export { H2 } from './typography/h2';
export { H3 } from './typography/h3';
export { H4 } from './typography/h4';
export { Text } from './typography/text';
export { Heading } from './typography/heading';

export { Button } from './button';
export { ButtonGroup } from './button-group';
export { FieldHint } from './field-hint';
export { Icon } from './icon';
export { Input } from './input';
export { List } from './list';
export { LoadingSpinner } from './loading-spinner';
export { Radio } from './radio';
export { StarRating } from './star-rating';
export { StepMarks } from './step-marks';
export { ToggleButton } from './toggle-button';
