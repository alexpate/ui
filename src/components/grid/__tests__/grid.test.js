/* @flow */
/* eslint-env jest */
import * as React from 'react';
import {sheet} from 'emotion';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';
import {Grid, Row, Column} from 'components/grid';

expect.addSnapshotSerializer(serializer(sheet));

describe('components/grid', () => {
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <Grid {...props}>
            <Row>
              <Column columns={2} />
              <Column columns={10} />
            </Row>
          </Grid>
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });
});
