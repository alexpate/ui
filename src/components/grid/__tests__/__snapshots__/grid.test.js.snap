// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid should render shallow component ok 1`] = `
<div
  className="container"
  css={Object {}}
>
  <div
    className="row"
    css={Object {}}
  >
    <div
      className="col-xs-2"
      css={Object {}}
    />
    <div
      className="col-xs-10"
      css={Object {}}
    />
  </div>
</div>
`;
